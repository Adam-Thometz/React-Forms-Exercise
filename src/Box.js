import React from "react";
import './Box.css'

const Box = ({width, height, color, id, handleRemove}) => {
  const remove = () => handleRemove(id)
  return (
    <div className="Box" id={id}>
      <div style={{
        backgroundColor: color,
        width: `${width}px`,
        height: `${height}px`
      }} data-testid="box">
        
      </div>
      <button onClick={remove}>Remove</button>
    </div>
  )
}

export default Box