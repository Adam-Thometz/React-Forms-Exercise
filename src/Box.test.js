import { render, screen } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', () => {
  render(<Box />);
});

it('matches snapshot', () => {
  const {asFragment} = render(<Box color="blue" height="100" width="200" />)
  expect(asFragment()).toMatchSnapshot()
});