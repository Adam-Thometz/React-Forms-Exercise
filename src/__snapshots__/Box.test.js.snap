// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      data-testid="box"
      style="background-color: blue; width: 200px; height: 100px;"
    />
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
